pkg/bgpv1/test/adverts_test.go

```diff
@@ -622,3 +627,188 @@ func Test_LBEgressAdvertisement(t *testing.T) {
 		})
 	}
 }
+
+// Test_AdvertisedPathAttributes validates optional path attributes in advertised paths.
+func Test_AdvertisedPathAttributes(t *testing.T) {
+	testutils.PrivilegedTest(t)
+
+	node.SetTestLocalNodeStore()
+	defer node.UnsetTestLocalNodeStore()
+
+	var steps = []struct {
+		description         string
+		op                  string // add or update
+		podCIDRs            []string
+		lbService           *lbSrvConfig
+		lbPool              *lbPoolConfig
+		advertiseAttributes []v2alpha1.CiliumBGPPathAttributes
+		expectedRouteEvent  routeEvent
+	}{
+		{
+			description: "advertise pod CIDR with standard community + non-default local pref",
+			op:          "add",
+			podCIDRs:    []string{"10.1.0.0/16"},
+			advertiseAttributes: []v2alpha1.CiliumBGPPathAttributes{
+				{
+					SelectorType: v2alpha1.PodCIDRSelectorName,
+					Communities: &v2alpha1.BGPCommunities{
+						Standard: []v2alpha1.BGPStandardCommunity{v2alpha1.BGPStandardCommunity("64125:100")},
+					},
+					LocalPreference: pointer.Int64(150),
+				},
+			},
+			expectedRouteEvent: routeEvent{
+				sourceASN:   ciliumASN,
+				prefix:      "10.1.0.0",
+				prefixLen:   16,
+				isWithdrawn: false,
+				extraPathAttributes: []bgp.PathAttributeInterface{
+					bgp.NewPathAttributeLocalPref(150),
+					bgp.NewPathAttributeCommunities([]uint32{parseCommunity("64125:100")}),
+				},
+			},
+		},
+		{
+			description: "advertise service IP with large community",
+			op:          "add",
+			lbService: &lbSrvConfig{
+				name:      "service-a",
+				ingressIP: "10.100.1.111",
+			},
+			lbPool: &lbPoolConfig{
+				name:  "pool-a",
+				cidrs: []string{"10.100.1.0/24"},
+			},
+			advertiseAttributes: []v2alpha1.CiliumBGPPathAttributes{
+				{
+					SelectorType: v2alpha1.CiliumLoadBalancerIPPoolSelectorName,
+					Communities: &v2alpha1.BGPCommunities{
+						Large: []v2alpha1.BGPLargeCommunity{v2alpha1.BGPLargeCommunity("64125:100:200")},
+					},
+				},
+			},
+			expectedRouteEvent: routeEvent{
+				sourceASN:   ciliumASN,
+				prefix:      "10.100.1.111",
+				prefixLen:   32,
+				isWithdrawn: false,
+				extraPathAttributes: []bgp.PathAttributeInterface{
+					bgp.NewPathAttributeLocalPref(100),
+					bgp.NewPathAttributeLargeCommunities([]*bgp.LargeCommunity{
+						{
+							ASN:        64125,
+							LocalData1: 100,
+							LocalData2: 200,
+						},
+					}),
+				},
+			},
+		},
+	}
+
+	testCtx, testDone := context.WithTimeout(context.Background(), maxTestDuration)
+	defer testDone()
+
+	// setup topology - iBGP (ASN == ciliumASN)
```

Comment:
Why use iBGP for this test? Are some of these communities non-transitive between AS?

Reply:
To test Local Preference. That is not advertised to eBGP peers.